src/libs/fileDownload/FileUtils.js

```diff
@@ -0,0 +1,112 @@
+import {Alert, Linking} from 'react-native';
+import moment from 'moment';
+import CONST from '../../CONST';
+import * as Localize from '../Localize';
+
+/**
+ * Show alert on successful attachment download
+ */
+function showSuccessAlert() {
+    Alert.alert(
+        Localize.translateLocal('fileDownload.success.title'),
+        Localize.translateLocal('fileDownload.success.message'),
+        [
+            {
+                text: Localize.translateLocal('fileDownload.success.buttonOkText'),
+                style: 'cancel',
+            },
+        ],
+        {cancelable: false},
+    );
+}
+
+/**
+ * Show alert on attachment download error
+ */
+function showGeneralErrorAlert() {
+    Alert.alert(
+        Localize.translateLocal('fileDownload.generalError.title'),
+        Localize.translateLocal('fileDownload.generalError.message'),
+        [
+            {
+                text: Localize.translateLocal('fileDownload.generalError.buttonCancelText'),
+                style: 'cancel',
+            },
+        ],
+    );
+}
+
+/**
+ * Show alert on attachment download permissions error
+ */
+function showPermissionErrorAlert() {
+    Alert.alert(
+        Localize.translateLocal('fileDownload.permissionError.title'),
+        Localize.translateLocal('fileDownload.permissionError.message'),
+        [
+            {
+                text: Localize.translateLocal('fileDownload.permissionError.buttonCancelText'),
+                style: 'cancel',
+            },
+            {
+                text: Localize.translateLocal('fileDownload.permissionError.buttonSettingsText'),
+                onPress: () => Linking.openSettings(),
+            },
+        ],
+    );
+}
+
+/**
+ * Generate a random file name with timestamp and file extension
+ * @param {String} url
+ * @returns {String}
+ */
+function getAttachmentName(url) {
+    if (!url) {
+        return '';
+    }
+    return `${moment().format('DDMMYYYYHHmmss')}.${url.split(/[#?]/)[0].split('.').pop().trim()}`;
+}
+
+/**
+ * @param {String} fileName
+ * @returns {Boolean}
+ */
+function isImage(fileName) {
+    return CONST.FILE_TYPE_REGEX.IMAGE.test(fileName);
+}
+
+/**
+ * @param {String} fileName
+ * @returns {Boolean}
+ */
+function isVideo(fileName) {
+    return CONST.FILE_TYPE_REGEX.VIDEO.test(fileName);
+}
+
+/**
+ * Returns file type based on the uri
+ * @param {String} fileUrl
+ * @returns {String}
+ */
+function getFileType(fileUrl) {
+    if (!fileUrl) {
+        return;
+    }
+    const fileName = fileUrl.split('/').pop().split('?')[0].split('#')[0];
```

Comment:
Would be needed as our URLs have query strings`?` and `#` and we would want to look at the file name only. I can now remove the `split('/').pop()`. Want me to remove or keep it?

Reply:
I understand the `$` usage. What I was trying to explain is:
```

http://localhost/abc/def/geh/assets/images/image.jpg // This will match

http://localhost/abc/def/geh/assets/images/image.jpg?encryptedToken=jwhfjhewhfwk // This won't 
```

and our image paths are with a query string (?). As I asked earlier, I can remove the `split('/').pop()` if you feel it adds any value. Please let me know if you have any further questions.

Reply:
Ah, I missed that this is public URLs. I thought we are getting local URLs only. So it would be better to use URL constructor to get the pathname instead of this manual string manipulation.

Reply:
> So it would be better to use URL constructor

Agree with this.

Reply:
Does it work with react native? I tried and it says `URL.pathname` is not implemented.  I can't see any implementation in our codebase. Do you want me to add a polyfill?

Reply:
I think it is not implemented so yeah let's keep it. But please test it thoroughly with many URLs encoded or decode both.